src/Microsoft.VisualStudio.Services.Agent/JobServerQueue.cs

```diff
@@ -640,10 +653,20 @@ private async Task UploadFile(UploadFileInfo file)
                 }
                 else
                 {
-                    // Create attachment
-                    using (FileStream fs = File.Open(file.Path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+                    if (_writeToBlobStoreAttachments)
```

Comment:
Should we consider a fallback on attachments the way we have  fallback on logs?

Reply:
I was debating it. We did it for logs but not for artifacts. These are more like artifacts... but the cost isn't that high so why not?

Reply:
+1 on this - that bailed us out a few times with our rollout